File: 2147-number-of-ways-to-divide-a-long-corridor/2147-number-of-ways-to-divide-a-long-corridor.js
/**
 * @param {string} corridor
 * @return {number}
 */
var numberOfWays = function(corridor) {
    let mod = 1e9 + 7
    let seatS = []
    for(let i = 0; i < corridor.length; i++) {
        if(corridor[i] == 'S')
            seatS.push(i)
    }
    if(seatS.length % 2 != 0 || seatS.length == 0)
        return 0
    let result = 1
    for(let i = 2; i < seatS.length; i += 2) {
        result = (result * (seatS[i] - seatS[i-1])) % mod
    }
    return result
};

File: 2147-number-of-ways-to-divide-a-long-corridor/2147-number-of-ways-to-divide-a-long-corridor.cpp
class Solution {
public:
    int numberOfWays(string corridor) {
        int n= corridor.length();
        int ans=1;
        int totalSeats=0;
        const int mod = 1e9+7;
        for(int i=0; i<n; i++){
            if(corridor[i]=='S'){
                totalSeats++;
            }
        }

        // if totalSeats is odd 
        if(totalSeats==0 || totalSeats%2!=0) return 0;

        int CurrSeatCount =0, plants=0;
      for(auto c: corridor){
        if(c=='S'){
           CurrSeatCount++;
           if(CurrSeatCount>2 && CurrSeatCount%2==1){
            ans = (1LL*ans*(plants+1))%mod;
            plants=0;
           }
        }
        else{
            // not all P matter only after seat pairs matter
            if(CurrSeatCount>=2 && CurrSeatCount%2==0){
                plants++;
            }
        }
      }
        return ans;
    }
};